crates/runtime/src/dataconnector/kafka.rs

```diff
@@ -355,36 +391,44 @@ async fn bootstrap_new_kafka_consumer(
             connector_component: ConnectorComponent::from(dataset),
         })?;
 
-    let msg = match kafka_consumer
-        .next_json::<serde_json::Value, serde_json::Value>()
-        .await
-    {
-        Ok(Some(msg)) => msg,
-        Ok(None) => {
-            return Err(super::DataConnectorError::UnableToGetReadProvider {
-                dataconnector: "kafka".to_string(),
-                source: "No message received from Kafka.".into(),
-                connector_component: ConnectorComponent::from(dataset),
-            });
-        }
-        Err(e) => {
-            return Err(e).boxed().context(super::UnableToGetReadProviderSnafu {
-                dataconnector: "kafka",
-                connector_component: ConnectorComponent::from(dataset),
-            });
+    let schema_inference_sample_count = schema_inference_sample_count.unwrap_or(1);
+
+    // Read schema_inference_sample_count messages to infer schema
+    // this is useful when some of the fields could be optional and use 'null'
+    let mut sample_values = Vec::with_capacity(schema_inference_sample_count);
+
+    for _ in 0..schema_inference_sample_count {
+        match kafka_consumer
+            .next_json::<serde_json::Value, serde_json::Value>()
+            .await
+        {
+            Ok(Some(msg)) => sample_values.push(msg),
+            Ok(None) => {
```

Comment:
If I set `schema_inference_sample_count=100`, but the topic has not yet had 100 messages, we expect to fail?

Reply:
Or `Ok(None)` just occurs if topic/connection is closed?

Reply:
My understanding is that `Ok(None)` can only be reported when connection/topic is closed. Otherwise consumer/connector just waits for more data available to complete sampling / schema inference